simulator/src/main/java/com/hedera/block/simulator/grpc/GrpcInjectionModule.java

```diff
@@ -32,6 +34,16 @@ public interface GrpcInjectionModule {
      */
     @Singleton
     @Binds
-    PublishStreamGrpcClient bindPublishStreamGrpcClient(
-            PublishStreamGrpcClientImpl publishStreamGrpcClient);
+    PublishStreamGrpcClient bindPublishStreamGrpcClient(PublishStreamGrpcClientImpl publishStreamGrpcClient);
+
+    /**
+     * Provides the stream enabled flag
+     *
+     * @return the stream enabled flag
+     */
+    @Singleton
+    @Provides
+    static AtomicBoolean provideStreamEnabledFlag() {
+        return new AtomicBoolean(true);
+    }
```

Comment:
I am not quite sure if we should ever declare and DI core java classes like that. Maybe a better approach would be to do `@Provides` at the place where this `AtomicBoolean` must be injected (I believe this is the same file at line 37) and instantiate it there, while getting the rest of the `dagger` DI enabled dependencies as method arguments.

Reply:
I believe we are just seeing the _incomplete_ nature of the system at the moment, here.  This shouldn't be a single `AtomicBoolean` long term, that's just a temporary expedient.  Longer term we'll almost certainly have some sort of "stream control" object that includes a number of different control attributes and we'll replace this single value with that more complete object.

Reply:
@jsync-swirlds I do agree that the ticket aims to be a temporary fix and not the final solution. That being said, I did also aim to make my comments based on that.